src/main/java/com/example/spring/service/cart/CartServiceImpl.java

```diff
@@ -0,0 +1,75 @@
+package com.example.spring.service.cart;
+
+import com.example.spring.dto.cartitem.CartItemDto;
+import com.example.spring.dto.cartitem.CartItemRequestDto;
+import com.example.spring.dto.cartitem.UpdateCartItemRequestDto;
+import com.example.spring.dto.shoppingcart.ShoppingCartDto;
+import com.example.spring.exception.EntityNotFoundException;
+import com.example.spring.mapper.CartItemMapper;
+import com.example.spring.mapper.ShoppingCartMapper;
+import com.example.spring.model.CartItem;
+import com.example.spring.model.ShoppingCart;
+import com.example.spring.repository.cartitem.CartItemRepository;
+import com.example.spring.repository.shoppingcart.ShoppingCartRepository;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CartServiceImpl implements CartService {
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    public ShoppingCartDto save(String email, CartItemRequestDto cartItemRequestDto) {
+        CartItem cartItem = cartItemMapper.toModel(cartItemRequestDto);
+        ShoppingCart shoppingCart = findByUser(email);
+        shoppingCart.getCartItems().add(cartItem);
+        cartItem.setShoppingCart(shoppingCart);
+        cartItemRepository.save(cartItem);
+        return shoppingCartMapper.toDto(shoppingCart);
+    }

```

Comment:
It would be best if you made it method Transactional. fix the same issue in all places